hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/store/contracts/precompile/TransferPrecompilesStubTest.java

```diff
@@ -0,0 +1,586 @@
+        given(impliedTransfersMarshal.validityWithCurrentProps(any())).willReturn(OK);
+        given(impliedTransfersMarshal.assessCustomFeesAndValidate(anyInt(), anyInt(), anyInt(), any(), any(), any()))
+                .willReturn(impliedTransfers);
+        // This test does not cover key activation. Rather a parameter is passed in to indicate pass/fail.
+        keyActivationUtils
+                .when(() -> validateKey(any(), any(), any(), any(), any(), eq(CryptoTransfer)))
+                .thenReturn(passesKeyActivation);
+
+        subject = new TransferPrecompile(
+                ledgers,
+                worldUpdater,
+                sigsVerifier,
+                sideEffects,
+                syntheticTxnFactory,
+                infrastructureFactory,
+                precompilePricingUtils,
+                selector,
+                senderAddress,
+                htsUnsupportedCustomFeeReceiverDebits, // This is passed in as a parameter
+                true,
+                false,
+                canFallbaToApprovals);
+
+        if (bodyMethodShouldPass) {
+            subject.body(inputBytes, a -> a);
+        } else {
+            assertThrows(IllegalArgumentException.class, () -> subject.body(inputBytes, a -> a));
+            return;
+        }
+
+        if (runMethodShouldPass) {
+            subject.run(frame);
+        } else {
+            assertThrows(InvalidTransactionException.class, () -> subject.run(frame));
+        }
+    }
+
+    private static Stream<Arguments> provideTransferPrecompileArgs() {
+        Set<CustomFeeType> allTypes = EnumSet.of(CustomFeeType.FIXED_FEE, CustomFeeType.ROYALTY_FALLBACK_FEE);
+        Set<CustomFeeType> noType = Collections.emptySet();
+        Set<CustomFeeType> fixedFeeType = EnumSet.of(CustomFeeType.FIXED_FEE);
+        Set<CustomFeeType> royaltyFallbackFeeType = EnumSet.of(CustomFeeType.ROYALTY_FALLBACK_FEE);
+        return Stream.of(
+                // These tests should pass - as indicated by the first argument
+                // Basic tests
+                Arguments.of(
+                        "basic transferToken no custom fee types allowed",
+                        true,
+                        true,
+                        ABI_ID_TRANSFER_TOKEN,
+                        TRANSFER_TOKEN_INPUT,
+                        getBasicFungibleChanges(),
+                        allTypes,
+                        true,
+                        false),
+                Arguments.of(
+                        "basic transferToken all custom fee types allowed ",
+                        true,
+                        true,
+                        ABI_ID_TRANSFER_TOKEN,
+                        TRANSFER_TOKEN_INPUT,
+                        getBasicFungibleChanges(),
+                        noType,
+                        true,
+                        false),
+                Arguments.of(
+                        "basic transferTokens",
```

Comment:
better name for this test case? (i.e. more specific?) (also below at L321 & L331)

also in naming these tests - some are specifically named "...NFT..." so obviously deal with NFTs.  the rest i guess deal with fungibles ("FT"?) - should that be made explicit in the name?